notion-core/src/test/scala/com/ambiata/notion/core/ExecutionLocationSpec.scala

```diff
@@ -0,0 +1,16 @@
+package com.ambiata.notion.core
+
+import org.scalacheck._, Arbitrary._
+import Arbitraries._
+import org.specs2._
+
+class ExecutionLocationSpec extends Specification with ScalaCheck { def is = s2"""
+
+ An Execution location can be turned into
+  a location ${prop { location: Location =>
+    val executionLocation = ExecutionLocation.fromLocation(location)
+    executionLocation.map(_.location) must beSome(location) or beNone
```

Comment:
Not that it matters, but the `or` looks funny here, would probably have done:

    executionLocation.map(_.location) ==== executionLocation.as(location)